--- one_instruction.desc.c	2018-01-01 00:00:00.000000000 +0100
+++ one_instruction.desc.c	2018-01-01 00:00:00.000000000 +0100
@@ -36,6 +36,8 @@
 #endif
 
     switch(opcode) {
+      case a: {/* This is a's code */}
+        break;
       default:
 	fprintf(stderr, "Undefined bytecode %d\n", opcode);
         break;
@@ -46,6 +48,7 @@
 struct ite {
   char *name; char *param; int val;
 } IT[] = {
+  { "a", "This is a", 0 } ,
 };
 
 void disassemble(unsigned int ptr, int len, FILE *f)
--- one_instruction.desc.h	2018-01-01 00:00:00.000000000 +0100
+++ one_instruction.desc.h	2018-01-01 00:00:00.000000000 +0100
@@ -0,0 +1 @@
+#define a 0
--- one_instruction.desc_i.h	2018-01-01 00:00:00.000000000 +0100
+++ one_instruction.desc_i.h	2018-01-01 00:00:00.000000000 +0100
@@ -52,7 +52,7 @@
 
 
 (null)
-#define NINSTRUCTIONS 0
+#define NINSTRUCTIONS 1
 typedef struct State {
   u32 firstThing;
   u32 pc;
